include/arch/arc/v2/linker.ld

```diff
@@ -4,9 +4,231 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+/**
+ * @brief Common parts of the linker scripts for the ARCv2 targets for
+ * GNU and MWDT toolchains.
+ */
+
+#include <linker/sections.h>
+#include <linker/linker-defs.h>
+#include <linker/linker-tool.h>
+
+/* physical address of RAM */
+#ifdef CONFIG_HARVARD
+	#define ROMABLE_REGION ICCM
+	#define RAMABLE_REGION DCCM
+#else
+	#if defined(CONFIG_XIP) && (FLASH_SIZE != 0)
+		#define ROMABLE_REGION FLASH
+		#define RAMABLE_REGION SRAM
+	#else
+		#define ROMABLE_REGION SRAM
+		#define RAMABLE_REGION SRAM
+	#endif
+#endif
+
+#ifdef CONFIG_ARC_MPU_ENABLE
+	#if CONFIG_ARC_MPU_VER == 2
+		#define MPU_MIN_SIZE 2048
+	#elif CONFIG_ARC_MPU_VER == 3
+		#define MPU_MIN_SIZE 32
+	#endif
+	#define MPU_MIN_SIZE_ALIGN . = ALIGN(MPU_MIN_SIZE);
+	#if defined(CONFIG_MPU_REQUIRES_POWER_OF_TWO_ALIGNMENT)
+		#define MPU_ALIGN(region_size) \
+			. = ALIGN(MPU_MIN_SIZE); \
+			. = ALIGN( 1 << LOG2CEIL(region_size))
+	#else
+		#define MPU_ALIGN(region_size) \
+			. = ALIGN(MPU_MIN_SIZE)
+	#endif
+#else
+	#define MPU_MIN_SIZE_ALIGN
+	#define MPU_ALIGN(region_size) . = ALIGN(4)
+#endif
 
-#if defined(__MWDT_LINKER_CMD__)
-#include <arch/arc/v2/linker-mwdt.ld>
+#if defined(CONFIG_XIP)
+	#define _DATA_IN_ROM __data_rom_start
 #else
-#include <arch/arc/v2/linker-gnu.ld>
+	#define _DATA_IN_ROM
 #endif
+
+OUTPUT_ARCH(arc)
+ENTRY(CONFIG_KERNEL_ENTRY)
+
+MEMORY {
+#ifdef FLASH_START
+	FLASH (rx)  : ORIGIN = FLASH_START, LENGTH = FLASH_SIZE
+#endif
+#ifdef ICCM_START
+	ICCM  (rwx)  : ORIGIN = ICCM_START,  LENGTH = ICCM_SIZE
+#endif
+#ifdef SRAM_START
+	SRAM  (rwx) : ORIGIN = SRAM_START,  LENGTH = SRAM_SIZE
+#endif
+#ifdef DCCM_START
+	DCCM  (rw)  : ORIGIN = DCCM_START,  LENGTH = DCCM_SIZE
+#endif
+	/* Used by and documented in include/linker/intlist.ld */
+	IDT_LIST  (wx)      : ORIGIN = 0xFFFFF7FF, LENGTH = 2K
+}
+
+SECTIONS {
+
+#include <linker/rel-sections.ld>
+
+	GROUP_START(ROMABLE_REGION)
+
+	SECTION_PROLOGUE(_TEXT_SECTION_NAME,,ALIGN(1024)) {
+		_image_rom_start = .;
+		_image_text_start = .;
+
+/* Located in generated directory. This file is populated by the
+ * zephyr_linker_sources() Cmake function.
+ */
+#include <snippets-rom-start.ld>
+		*(.text .text.*)
+		*(.gnu.linkonce.t.*)
+
+	. = ALIGN(4);
+#include <linker/kobject-text.ld>
+	} GROUP_LINK_IN(ROMABLE_REGION)
+
+	_image_text_end = .;
+	_image_rodata_start = .;
+
+#include <linker/common-rom.ld>
+
+	SECTION_PROLOGUE(_RODATA_SECTION_NAME,,) {
+		KEEP(*(.openocd_dbg))
+		KEEP(*(".openocd_dbg.*"))
+		*(".rodata")
+		*(".rodata.*")
+		*(.gnu.linkonce.r.*)
+
+/* Located in generated directory. This file is populated by the
+ * zephyr_linker_sources() Cmake function.
+ */
+#include <snippets-rodata.ld>
+
+#include <linker/kobject-rom.ld>
+	} GROUP_LINK_IN(ROMABLE_REGION)
+
+#ifdef __MWDT_LINKER_CMD__
+/* TODO: add mwdt specific ROM C++ sections */
```

Comment:
If for mwdt something special is required, please create an issue so we don't forget to take care of this.
(or, fix it...)